### ISSUE TITLE
Group-contract fails with cxf-codegen-plugin

### ISSUE THREAD

**Issue by waldeinburg** [mklemm/jaxb2-rich-contract-plugin#59]
When using group-contracts with Maven [cxf-codegen-plugin](https://cxf.apache.org/docs/maven-cxf-codegen-plugin-wsdl-to-java.html#Mavencxf-codegen-plugin(WSDLtoJava)-Example7:Usingxjcextensions) the build fails with the following `BadCommandLineException` (included below with the first line of the stack trace with `-X` Maven option set):

```
Error setting up group-interface-plugin javax.xml.xpath.XPathExpressionException: java.io.FileNotFoundException: /path/to/working/directory/null (No such file or directory)
    at com.kscs.util.plugins.xjc.GroupInterfacePlugin.generateDummyGroupUsages (GroupInterfacePlugin.java:218)
```

The plugin seems to collide with a hack in [`org.apache.cxf.tools.wsdlto.databinding.jaxb.JAXBDataBinding`](https://github.com/apache/cxf/blob/master/tools/wsdlto/databinding/jaxb/src/main/java/org/apache/cxf/tools/wsdlto/databinding/jaxb/JAXBDataBinding.java#L404):

```java
        if (context.get(ToolConstants.CFG_NO_ADDRESS_BINDING) == null
            || context.get(ToolConstants.CFG_XJC_ARGS) != null) {
            try {
                // keep parseArguments happy, supply dummy required command-line
                // opts
                opts.addGrammar(new InputSource("null"));
```

The `CFG_NO_ADDRESS_BINDING` value is null and the `CFG_XJC_ARGS` is of course not null because it contains `-Xgroup-contract`. This means that `GroupInterfacePlugin::generateDummyGroupUsages` will receive one input source, the dummy "null" value, and then findIncludeMappings will crash.

This is a quite puzzling hack. It's not that there's no input schemas; everything builds fine when removing groupt-contract from the plugin list (leaving other arguments to trig the conditional above). Will you implement a workaround or should I report this as a bug to Apache CXF, trying to make them make another workaround for the supposed issue with parseArgument?

**Comment by aSemy** [comment]
I have the same problem. Thank you for finding out the cause. 

I've created an issue https://issues.apache.org/jira/browse/CXF-8580

Do you know if there's a workaround?

**Comment by mklemm** [comment]
Should be looking for an alternative way to determine the output path of generated group interfaces...

**Comment by mklemm** [comment]
The root cause seems to be than an InputSource passed to an XPathExpression MUST have a valid SystemID pointing to a file or URI where the content is located. The WSDL tool spoils that.

**Comment by mklemm** [comment]
Fixed in 4.0.1-SNAPSHOT
